README.md

```diff
@@ -27,6 +27,46 @@ yarn build
 
 Open an issue to request a feature or report a bug.
 
+## Testing
+
+We use [Vitest](https://vitest.dev/) as the test runner.
+
+### Unit Tests
+
+```bash
+yarn install    # Install project dependencies
+
+yarn test:unit  # Run all tests
+```
+
+### End-to-End Tests
+
+End-to-end tests simulate an actual blockchain. Make sure you have [Docker](https://docs.docker.com/engine/install/) installed to run these tests.
+
+#### 1. Start the Blockchain Node
+
+We use a fork of [agoric-3](https://github.com/Agoric/agoric-sdk/releases) for our tests. To start the simulation, run the following command which pulls and runs the [`agoric-3-proposals`](https://github.com/Agoric/agoric-3-proposals/) image, setting up an environment similar to the Agoric mainnet:
```

Comment:
I'm not entirely clear whether you mean to refer to the `agoric-3` chain or to some release or releases. Assuming you mean the chain...

```suggestion
We use a simulation of `agoric-3` for our tests. To start the simulation, run the following command which pulls and runs the [`agoric-3-proposals`](https://github.com/Agoric/agoric-3-proposals/) image, setting up an environment similar to the Agoric mainnet:
```